### ISSUE TITLE
Unable to login to admin interface intermittently, NullPointerException when fail

### ISSUE THREAD

**Issue by mbjones** [NCEAS/metacat#1124]
---

Author Name: **Bryce Mecum** (Bryce Mecum)
Original Redmine Issue: 7062, https://projects.ecoinformatics.org/ecoinfo/issues/7062
Original Date: 2016-07-22
Original Assignee: Jing Tao

---

This has been happening for months. When I log into the Metacat admin panel /{context}/admin, I enter my LDAP credentials and click "Login". Sometimes this takes ~10 seconds (why?) but sometimes it takes minutes and then returns with an error:

LoginAdmin.authenticateUser - Could not log in as: uid=mecum,ou=Account,dc=ecoinformatics,dc=org : Connection to the authentication service failed in AuthSession.authenticate: AuthLdap.authenticate - NullPointerException while authenticating in AuthLdap.authenticate: java.lang.NullPointerException . Please try again 

I'll try again and it will go through without error.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Bryce Mecum** (Bryce Mecum)
Original Date: 2016-07-22T18:47:48Z

---

full trace:

metacat 20160722-11:46:01: [ERROR]: AuthLdap.authenticate - NullPointerException while authenticating in AuthLdap.authenticate: java.lang.NullPointerException [edu.ucsb.nceas.metacat.AuthLdap]
java.lang.NullPointerException
        at com.sun.jndi.ldap.LdapReferralContext.search(LdapReferralContext.java:652)
        at com.sun.jndi.ldap.LdapReferralContext.search(LdapReferralContext.java:644)
        at edu.ucsb.nceas.metacat.AuthLdap.getIdentifyingName(AuthLdap.java:486)
        at edu.ucsb.nceas.metacat.AuthLdap.authenticate(AuthLdap.java:159)
        at edu.ucsb.nceas.metacat.AuthSession.authenticate(AuthSession.java:93)
        at edu.ucsb.nceas.metacat.util.AuthUtil.logUserIn(AuthUtil.java:206)
        at edu.ucsb.nceas.metacat.admin.LoginAdmin.authenticateUser(LoginAdmin.java:113)
        at edu.ucsb.nceas.metacat.admin.MetacatAdminServlet.handleGetOrPost(MetacatAdminServlet.java:177)
        at edu.ucsb.nceas.metacat.admin.MetacatAdminServlet.doPost(MetacatAdminServlet.java:82)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:646)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:727)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:501)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:98)
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:950)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:408)
        at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:193)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:607)
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:313)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
metacat 20160722-11:46:01: [ERROR]: LoginAdmin.authenticateUser - Could not log in as: uid=mecum,ou=Account,dc=ecoinformatics,dc=org : <?xml version="1.0" encoding="UTF-8"?>
<error_login>
  <message>Connection to the authentication service failed in AuthSession.authenticate: AuthLdap.authenticate - NullPointerException while authenticating in AuthLdap.authenticate: java.lang.NullPointerException</message>
</error_login>. Please try again [edu.ucsb.nceas.metacat.admin.LoginAdmin]

**Comment by amoeba** [comment]
I haven't seen this in a very long time and I didn't provide quite enough detail for anyone to really reproduce. This was back when I was newer to Java and metacat so I might have been doing something out of order. I also remember we used to have some LDAP instance in our tree (is that the right lingo?) that would time out. We've since removed it so maybe this was that.

I'm going to close this for now since I can't reproduce.
